src/components/inputField/selectElement/SelectElement.tsx

```diff
@@ -0,0 +1,111 @@
+import { useEffect, useRef, useState } from "react";
+import CarretDownIcon from "../../../assets/images/icon-caret-down.svg?react";
+import "../inputField.scss";
+
+interface selectProps {
+    dropdownOpened: boolean;
+    selectDecorator:string | undefined;
+}
+const colorTags = [
```

Comment:
If you convert this to an object or a Map with the keys with what's now in the array and the values the display value you save a lot of lines of code and simplify the implementation.